### ISSUE TITLE
Error: error decoding response body: missing field `user_id` at line 1 column 69"

### ISSUE THREAD

**Issue by carboncls** [talaia-labs/rust-teos#109]
Hi! I've been using python-teos and recently switched to rust-teos.

I also upgraded from Ubuntu 20.04 to 22.04

Now, when using the watchtower-plugin for CLN, when registering with my own tower on the same machine, I get:

```
lightning-cli registertower 028690a1f58c734cd9c97584a79d81ac3b817db863906d6852cc49e26026fc4cc5@localhost
"Unexpected response body. Error: error decoding response body: missing field `user_id` at line 1 column 69"
```

What could be the issue here?

**Comment by meryacine** [comment]
The request made it to the receiver (the tower?) but the response wasn't the one expected. Is the tower running on that port (default port is 9814), not something else? Or maybe the tower failed to register for some reason and returned an erroneous response.

Try running:
```bash
curl -i -H "Accept: application/json" -H "Content-Type: application/json" -d '{"user_id": "03eee489cf70df7da8ac6254fce58b7e245b127022c15111edd856f8193ca75b51"}' -X POST localhost:9814/register
```
and share the tower's response.

**Comment by carboncls** [comment]
Ok, I managed to make it work. I think the issue was that when starting the daemon, it was python-teos starting.
I deleted the python-teos folder, teosd and teos-cli in /home/user/.local/bin/ and rebooted.
It all works now.

Is my tower now reachable for other people too? It runs with --torsupport.

**Comment by carboncls** [comment]
would love it if I can run `teosd` as a background daemon like python-teos.

**Comment by meryacine** [comment]
Yup, it is reachable using the onion url.

I thought about adding an issue for teosd to have the option to run as
daemon long time ago, but forgot to actually add it 😅.

Please feel free to post this issue.

**Comment by carboncls** [comment]
ok thanks!
I did open the issue =)

**Comment by sr-gi** [comment]
> Hi! I've been using python-teos and recently switched to rust-teos.
> 
> I also upgraded from Ubuntu 20.04 to 22.04
> 
> Now, when using the watchtower-plugin for CLN, when registering with my own tower on the same machine, I get:
> 
> ```
> lightning-cli registertower 028690a1f58c734cd9c97584a79d81ac3b817db863906d6852cc49e26026fc4cc5@localhost
> "Unexpected response body. Error: error decoding response body: missing field `user_id` at line 1 column 69"
> ```
> 
> What could be the issue here?

Looks like python and rustlang client / tower were being try to make work together. That won't work given we've changed some of the internals for how data is stored.

> Is my tower now reachable for other people too? It runs with --torsupport.

If you are running the code on master, then yes. We've recently added Tor support for the `watchtower-client` (the tower had it for quite a long time, but the CoreLN plugin didn't). This will be released with `v0.1.2`.
